File: scripts/diag_templates.py
import sqlite3, json

con = sqlite3.connect('dev.db')
con.row_factory = sqlite3.Row
cur = con.cursor()

print('=== ARCHITECTS ===')
architects = cur.execute('SELECT * FROM Architect').fetchall()
for a in architects:
    print(f"  id={dict(a)['id'][:8]}... firmName={dict(a)['firmName']}")

print()
print('=== TEMPLATES ===')
templates = cur.execute('SELECT * FROM Template').fetchall()
if not templates:
    print('  [NONE - template table is empty!]')
for t in templates:
    td = dict(t)
    pattern = json.loads(td['titleBlockPattern']) if td['titleBlockPattern'] else None
    print(f"  architectId={td['architectId'][:8]}...")
    print(f"  confirmedCount={pattern['confirmedDrawingCount'] if pattern else 'N/A'}")
    print(f"  side={pattern['side'] if pattern else 'N/A'}")
    print(f"  fieldPositions={'YES' if td['fieldPositions'] else 'NO'}")

print()
print('=== EXTRACTED DRAWINGS (sample 20) ===')
rows = cur.execute("""
  SELECT id, filename, extractionStatus, architectId, drawingNumber, titleBlockLocation, flags
  FROM Drawing 
  WHERE extractionStatus IN ('extracted','reviewed','published')
  LIMIT 20
""").fetchall()
print(f'Count: {len(rows)}')
for r in rows:
    rd = dict(r)
    flags = json.loads(rd['flags']) if rd['flags'] else []
    has_arch = rd['architectId'] is not None
    print(f"  {rd['filename'][:28]:28s} | drawingNum={str(rd['drawingNumber'] or ''):10s} | hasArch={has_arch} | loc={rd['titleBlockLocation']} | flags={','.join(flags)}")

print()
print('=== ARCHITECT LINKING SUMMARY ===')
c = cur.execute("SELECT COUNT(*) as cnt FROM Drawing WHERE architectId IS NULL").fetchone()
total = cur.execute("SELECT COUNT(*) as cnt FROM Drawing").fetchone()
print(f"  Drawings with no architect: {dict(c)['cnt']} / {dict(total)['cnt']}")

extracted_no_arch = cur.execute("""
  SELECT COUNT(*) as cnt FROM Drawing 
  WHERE architectId IS NULL AND extractionStatus IN ('extracted','reviewed','published')
""").fetchone()
print(f"  Extracted drawings with no architect: {dict(extracted_no_arch)['cnt']}")

con.close()

File: scripts/diag-templates.js
const path = require('path');
const { PrismaClient } = require(path.join(process.cwd(), 'src/generated/prisma'));
const p = new PrismaClient();

async function main() {
  // Check architects and templates
  const architects = await p.architect.findMany({
    include: {
      template: true,
      drawings: {
        select: {
          id: true,
          extractionStatus: true,
          architectId: true,
          drawingNumber: true,
          titleBlockLocation: true,
          flags: true,
        }
      }
    }
  });

  console.log('\n=== ARCHITECT TEMPLATE DIAGNOSTICS ===\n');
  for (const a of architects) {
    const extracted = a.drawings.filter(d => d.extractionStatus === 'extracted' || d.extractionStatus === 'reviewed' || d.extractionStatus === 'published');
    const withNumber = a.drawings.filter(d => d.drawingNumber);
    const pattern = a.template?.titleBlockPattern ? JSON.parse(a.template.titleBlockPattern) : null;

    console.log(`Architect: ${a.firmName} (${a.id})`);
    console.log(`  Template exists: ${!!a.template}`);
    console.log(`  Total drawings: ${a.drawings.length}`);
    console.log(`  Extracted drawings: ${extracted.length}`);
    console.log(`  Drawings with number: ${withNumber.length}`);
    if (pattern) {
      console.log(`  Pattern confirmedCount: ${pattern.confirmedDrawingCount}`);
      console.log(`  Pattern side: ${pattern.side}`);
      console.log(`  Pattern locked: ${pattern.confirmedDrawingCount >= 2}`);
    } else {
      console.log(`  No pattern stored`);
    }
    
    // Show flags for first 5 drawings
    for (const d of extracted.slice(0, 5)) {
      const flags = d.flags ? JSON.parse(d.flags) : [];
      console.log(`    Drawing ${d.id.slice(-6)}: number=${d.drawingNumber || 'NULL'}, flags=${flags.join(',') || 'none'}`);
    }
    console.log('');
  }

  // Check drawings without architect
  const drawingsNoArchitect = await p.drawing.count({ where: { architectId: null } });
  const drawingsTotal = await p.drawing.count();
  console.log(`=== DRAWINGS WITHOUT ARCHITECT: ${drawingsNoArchitect} / ${drawingsTotal} ===\n`);

  // Sample some drawings
  const samples = await p.drawing.findMany({
    take: 5,
    where: { architectId: null, extractionStatus: { in: ['extracted', 'reviewed'] }},
    select: { id: true, filename: true, drawingNumber: true, extractionStatus: true }
  });
  if (samples.length > 0) {
    console.log('Sample drawings with no architect:');
    for (const s of samples) {
      console.log(`  ${s.filename} - ${s.drawingNumber} (${s.extractionStatus})`);
    }
  }
}

main().catch(e => { console.error(e); process.exit(1); }).finally(() => p.$disconnect());
